rust/arrow/src/ipc/reader.rs

```diff
@@ -560,11 +667,42 @@ mod tests {
             ))
             .unwrap();
 
-            let mut reader = Reader::try_new(file).unwrap();
+            let mut reader = FileReader::try_new(file).unwrap();
+
+            // read expected JSON output
+            let arrow_json = read_gzip_json(path);
+            assert!(arrow_json.equals_reader(&mut reader));
+        });
+    }
+
+    #[test]
+    fn read_generated_streams() {
+        let testdata = env::var("ARROW_TEST_DATA").expect("ARROW_TEST_DATA not defined");
+        // the test is repetitive, thus we can read all supported files at once
+        let paths = vec![
+            "generated_interval",
+            "generated_datetime",
+            "generated_nested",
+            "generated_primitive_no_batches",
+            "generated_primitive_zerolength",
+            "generated_primitive",
+        ];
+        paths.iter().for_each(|path| {
+            let file = File::open(format!(
+                "{}/arrow-ipc/integration/0.14.1/{}.stream",
```

Comment:
Is there anywhere that we can read this version number (0.14.1) from?  Separate PR obviously.

Reply:
We can address this when adding Rust to the integration testing. I'm not sure of how other implementations handle this.